### PR TITLE
Modularize the controller and node services

### PR DESCRIPTION
**Pull request by hughdanliu**
**Is this a bug fix or adding new feature?**
Refactor for future features. Allows us to add flags to the controller/node services and adds support for JSON logging.
**What is this PR about? / Why do we need it?**
Modularizes the controller and node services. This refactor will allow us to support node start-up taints, driver service deployment options, and makes our implementation consistent with EBS/FSx for Lustre.
Also configures JSON logging and text (default) logging options for the CSI driver.

**What testing is done?** 
Ran unit tests via `make test` and ran sanity tests via `make test-sanity`
Performed various rounds of manual testing to ensure functionality was kept intact.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	json "k8s.io/component-base/logs/json"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/aws-fsx-openzfs-csi-driver/pkg/driver"
+)
+
+func main() {
+	fs := flag.NewFlagSet("aws-fsx-csi-driver", flag.ExitOnError)
+
+	if err := logsapi.RegisterLogFormat(logsapi.JSONLogFormat, json.Factory{}, logsapi.LoggingBetaOptions); err != nil {
```
Comment (jacobwolfaws) [inline cmd/main.go:31]: let's remove this and add make it a separate commit supporting json, like this one: https://github.com/kubernetes-sigs/aws-fsx-csi-driver/commit/e86f25e6fa8ccf9ec677e17d56c517072c276cb0 and also come of the changes in here:
https://github.com/kubernetes-sigs/aws-fsx-csi-driver/commit/770a2c6e95ec01e7c92c4899b838d2570841d26a

Comment (hughdanliu) [reply cmd/main.go:31]: done

**Comment by jacobwolfaws** [comment]
It doesn't seem like the deploy/.../node-daemonset.yaml was updated to include the logging line. Did you generate the deployment files with `make generate-kustomize`?
